src/server/game/Scripting/ScriptReloadMgr.cpp

```diff
@@ -67,6 +67,9 @@ namespace fs = boost::filesystem;
 #ifdef _WIN32
     #include <windows.h>
     #define HOTSWAP_PLATFORM_REQUIRES_CACHING
+#elif __APPLE__
+    #include <dlfcn.h>
+    #define HOTSWAP_PLATFORM_REQUIRES_CACHING
```

Comment:
How did you see that the scripts are still usable by the core?
Maybe you could explain your impressions in detail.

Caching is required only if you can't delete shared libraries out of the `scripts` directory through your file explorer (when caching is disabled).